src/browser-bookmarks/import-bookmarks.js

```diff
@@ -1,143 +1,148 @@
-        if (referringVisitItemId && referringVisitItemId !== '0') {
-            // Look up what visit this id maps to in our model.
-            const referringVisitDoc = visitDocs[referringVisitItemId]
-            if (referringVisitDoc) {
-                const referringVisitDocId = referringVisitDoc._id
-                // Add a reference to the visit document.
-                visitDoc.referringVisit = {_id: referringVisitDocId}
-            }
-            else {
-                // Apparently the referring visit is not present in the history.
-                // We can just pretend that there was no referrer.
-            }
+      // Take and forget the id of the visitItem in the browser's history.
+      const referringVisitItemId = visitDoc.referringVisitItemId
+      delete visitDoc.referringVisitItemId
+      if (referringVisitItemId && referringVisitItemId !== '0') {
+        // Look up what visit this id maps to in our model.
+        const referringVisitDoc = visitDocs[referringVisitItemId]
+        if (referringVisitDoc) {
+          const referringVisitDocId = referringVisitDoc._id
+          // Add a reference to the visit document.
+          visitDoc.referringVisit = {_id: referringVisitDocId}
+        }
+        else {
+          // Apparently the referring visit is not present in the history.
+          // We can just pretend that there was no referrer.
         }
+      }
     })
     // Return the new docs.
     return {
-        pageDocs,
-        visitDocs: Object.values(visitDocs) // we can forget the old ids now
+      pageDocs,
+      visitDocs: Object.values(visitDocs) // we can forget the old ids now
     }
-}
-
-function logtree(bookmarkItem) {
-  if (bookmarkItem.url) {
-    console.log(makeIndent(indent) + bookmarkItem.url);
-  } else {
-    console.log(makeIndent(indent) + "Folder");
-    indent++;
   }
-  if (bookmarkItem.children) {
-    for (child of bookmarkItem.children) {
-      bookmarkItems.push(isWorthRemembering(bookmarkItem.url))
+
+  function logtree(bookmarkItem, indent) {
+    if (bookmarkItem.url) {
+      console.log(makeIndent(indent) + bookmarkItem.url);
+    } else {
+      console.log(makeIndent(indent) + "Folder");
+      indent++;
+    }
+    if (bookmarkItem.children) {
+      for (child of bookmarkItem.children) {
+        bookmarkItems.push(isWorthRemembering(bookmarkItem.url))
+      }
     }
+    return bookmarkItems;
   }
-  return bookmarkItems;
-}
 
-function onRejected(error) {
-  console.log(`An error: ${error}`);
-}
+  function onRejected(error) {
+    console.log(`An error: ${error}`);
+  }
 
-function getBookmarkItems() {
-  var bookmarkTree = browser.bookmarks.getTree();
-  return bookmarkTree.then(logtree(bookmarkTree[0]), onRejected);
-}
+  function getBookmarkItems() {
+    var bookmarkTree = browser.bookmarks.getRecent();
+    console.log("inside getBookmarkItems");
+    return bookmarkTree.then(logtree(bookmarkTree[0], 0), onRejected);
```

Comment:
`bookmarkTree` is a promise, it will give you the value only later. You should pass `then()` a callback function, which will be called with the result of the promise (the bookmarks in this case) as soon as that result is available.
Perhaps read a tutorial on javascript Promises, and have a look at other examples in the code.
Try something like `bookmarkTree.then(result => console.log(result), onRejected)`. In case you are not familiar with ES6, `arg => returnValue` is just short for `function (arg) {return returnValue}`.